packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/heatmap.dart

```diff
@@ -0,0 +1,384 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart'
+    show immutable, listEquals, objectRuntimeType;
+import 'package:flutter/material.dart' show Color;
+
+import 'types.dart';
+
+/// Uniquely identifies a [Heatmap] among [GoogleMap] heatmaps.
+///
+/// This does not have to be globally unique, only unique among the list.
+@immutable
+class HeatmapId extends MapsObjectId<Heatmap> {
+  /// Creates an immutable identifier for a [Heatmap].
+  const HeatmapId(super.value);
+}
+
+/// A wrapper around platform specific behavior for the radius of a [Heatmap].
+///
+/// Currently this class only handles platform-specific values, but in the
+/// future it may provide alternate constructors that abstract platform
+/// differences in handling of heatmap radius values. See flutter/flutter#145411.
+@immutable
+class HeatmapRadius {
+  /// Create a [HeatmapRadius] from a platform-independent [radius] value.
+  ///
+  /// In the future, this will convert the platform-independent [radius] to a
+  /// platform-specific value for [pixels].
+  // const HeatmapRadius(this.radius) : pixels = radius;
+
+  /// Create a [HeatmapRadius] with a radius in pixels.
+  ///
+  /// This value will be used verbatim on all platforms. It is up to the
+  /// developer to ensure that the value is appropriate for the platform.
```

Comment:
For my own eduction what does a developer need to do to know the value is appropriate? What is an example difference between android and ios?

Reply:
I thought @stuartmorgan already commented on this, but the issue is we aren't sure. Developers will need to test on each platform to find a value that is suitable.  Here is what the example project does which may or may not be ideal:
```dart
// The Google Maps SDK handles radius differently across platforms.
// It is the developer's responsibility to ensure proper behavior on
// each platform.
final int radius;
if (kIsWeb) {
  radius = 10;
} else {
  radius = switch (defaultTargetPlatform) {
    TargetPlatform.android => 20,
    TargetPlatform.iOS => 40,
    _ => 20,
  };
}
```

Reply:
I'm going to attempt re-writing this again